packages/flutter/lib/src/material/data_table.dart

```diff
@@ -64,6 +66,16 @@ class DataColumn {
   /// The label should not include the sort indicator.
   final Widget label;
 
+  /// {@template flutter.material.dataColumn.heading}
+  /// The heading string used for screen reader announcements.
+  ///
+  /// Announced by screen readers to help make sense of the information
+  /// in a [DataCell].
+  /// {@endtemplate}
+  ///
+  /// Summarizes the information displayed in this [DataColumn].
+  final String? heading;
```

Comment:
heading feels like something that will be displayed on ui. Since this is for semantics only, it should have the name semantics in it. If this mean to be a string representation of the `label` widget. consider change this to `semanticsLabel`
otherwise, some ideas: `semanticsTitle` `semanticsSummary`.
If the goal